src/yoroi-wallets/cardano/shelley/transactionCache.ts

```diff
@@ -505,6 +518,22 @@ export const makeTxCacheStorage = (storage: Storage): TxCacheStorage => ({
   },
 })
 
+const addMemos = async (transactions: Record<string, Transaction>, memosStorage: Storage) => {
+  const memosTxIds = await memosStorage.getAllKeys()
+
+  memosTxIds.forEach(async (memoTxId) => {
+    const memo = (await memosStorage.getItem(memoTxId)) as string
+    transactions = Object.assign(transactions, {
+      [memoTxId]: {
+        ...transactions[memoTxId],
+        memo,
+      },
+    })
+  })
```

Comment:
`Object.assign` modifies the object in the first param, so I would suggest using `{...transactions, [memoTxId]: ...}` or `Object.assign({}, transactions, {...})`